docs/rfcs/2023-09-12-remote-profile.md

```diff
@@ -0,0 +1,188 @@
+
+Therefore, the non-ephemeral part of handshake material should be reused for reconnects.
+
+TBD
+
+### Disposal
+
+The session may have a lifetime that a desktop or a mobile may stipulate while preparing a session.
+Alternatively a mobile (or a desktop, why not) may signal that they're done here and no further activity should be going with the session parameters.
+
+## Proposed UX flow
+
+### On a mobile device
+
+1. A user opens sidebar and clicks "connect from desktop" in the "You" section, starting remote controller discovery.
+  * UI enters "Waiting for desktop" window, which collects all the broadcasts received so far.
+  * Discovery process starts UDP broadcast listener on application port (5226).
+  * A datagram containing remote controller fingerprint is checked against a list of pre-registered controller devices.
+    - If the device is already known, the host establishes connection and UI transitions into "connection status" window.
+    - For unknown/new broadcasts a fingerprint is displayed instead.
+    - If the datagram contains no valid fingerprint, it is ignored.
+2. Clicking on unknown device fingerprint in the list starts OOB handshake.
+  * UI enters "New device" window, displaying a fingerprint and asking to scan a QR code (or paste a link, like in the contact screen).
+  * A OOB data from the QR/link contains remote controller fingerprint and remote display name, which is stored in device DB.
+    - Fingerprint must match the announce.
+    - Display name can be changed.
+  * Accepting the OOB automatically triggers remote controller connection and transitions UI to "connection status" window.
+3. A remote session initiated with a known device, or as a result of OOB exchange.
+  * A "connection status" window shows registered display name and current session status.
+  * Chat controller attempts to establish a remote session.
+    - A TCP connection is made to the address discovered.
+    - A TLS connection is instantiated, checking for remote CA fingerprint matches the previously established.
+    - A HTTP2 server is started on the mobile side of the TLS connection.
+    - The remote controller connects and subscribes for its output queue, marking the session established.
+  * For the duration of the remote session, the UI remains in the status window, preventing user interaction.
+    - This restriction may be lifted later.
+
+At any time a user may click on a "cancel" button and return to the main UI.
+
+In the "Network & servers" section of "Settings", there is an item to list all the registered remote controllers with buttons attached to *dispose* them one by one.
+*Disposing* a remote controller means its entry will be removed from database and any associated files deleted (there are none right now).
+Future connection attempts from a disposed device would be treated exactly as from a previously-unknown device.
```

Comment:
which means that desktop should be able to handle the scenario of connecting to known mobile that removed it - it should show qr code as a result of request from mobile.

Reply:
a desktop can always show a profile QR while waiting for connection.